recipes/msys2/all/conanfile.py

```diff
@@ -11,50 +11,91 @@ class MSYS2Conan(ConanFile):
     homepage = "http://www.msys2.org"
     license = "MSYS license"
     topics = ("conan", "msys", "unix", "subsystem")
-    build_requires = "7zip/19.00"
     short_paths = True
-    options = {"exclude_files": "ANY",  # Comma separated list of file patterns to exclude from the package
-               "packages": "ANY",  # Comma separated
-               "additional_packages": "ANY"}    # Comma separated
-    default_options = {"exclude_files": "*/link.exe",
-                       "packages": "base-devel,binutils,gcc",
-                       "additional_packages": None}
-    settings = "os_build", "arch_build"
-
-    def configure(self):
-        if self.settings.os_build != "Windows":
+    # "exclude_files" "packages" "additional_packages" values are a comma separated list
+    options = {
+        "exclude_files": "ANY",
+        "packages": "ANY",
+        "additional_packages": "ANY"
+    }
+    default_options = {
+        "exclude_files": "*/link.exe",
+        "packages": "base-devel,binutils,gcc",
+        "additional_packages": None
+    }
+    settings = "os", "arch"
+
+    def validate(self):
+        if self.settings.os != "Windows":
             raise ConanInvalidConfiguration("Only Windows supported")
-
-    def source(self):
-        # build tools have to download files in build method when the
-        # source files downloaded will be different based on architecture or OS
-        pass
+        if tools.Version(self.version) >= "20210105" and self.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration("Only Windows x64 supported")
 
     def _download(self, url, sha256):
         from six.moves.urllib.parse import urlparse
         filename = os.path.basename(urlparse(url[0]).path)
         tools.download(url=url, filename=filename, sha256=sha256)
         return filename
 
+    @property
+    def _keyring_subfolder(self):
+        return os.path.join(self.source_folder, "keyring")
+
+    @property
+    def _keyring_file(self):
+        return os.path.join(self._keyring_subfolder, "msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz")
+    
+    @property
+    def _keyring_sig(self):
+        return os.path.join(self._keyring_subfolder, "msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig")    
+
+    # download the new keyring
+    # https://www.msys2.org/news/#2020-06-29-new-packagers
+    def _download_keyring(self):
+        tools.download( "http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz",
+                        self._keyring_file,
+                        sha256="f1cc152902fd6018868b64d015cab9bf547ff9789d8bd7c0d798fb2b22367b2b" )
+        tools.download( "http://repo.msys2.org/msys/x86_64/msys2-keyring-r21.b39fb11-1-any.pkg.tar.xz.sig",
+                        self._keyring_sig,
+                        sha256="bbd22e88f33c81c40b145c34d8027d60f714d4fd1d0dccd456895f398cc56297" )
```

Comment:
Can this be added to the `conandata.yml`?
The hooks accept the following. 
```yml
sources:
  "1.0":
    - url: "https://www.example.com/"
      sha256: "abc"
    - url:
        keyring: "https://www.keyring.com"
      sha256: "abc"

```
(Not that I find it nice)

Reply:
Maybe.I just don't see any real advantage. The keys only needed for old packages, there is also no deference x86/x86_64.

Reply:
I understand re the keys.

About `x86/x86_64`, the recipe currently differentiates between these 2 archs by their index in a list.
Look at the `android-ndk` `conandata.yml` how the different operatings systems are handled.